src/hotspot/share/runtime/mutexLocker.hpp

```diff
@@ -239,16 +271,10 @@ class MonitorLocker: public MutexLocker {
 
  public:
   MonitorLocker(Monitor* monitor, Mutex::SafepointCheckFlag flag = Mutex::_safepoint_check_flag) :
-    MutexLocker(monitor, flag), _flag(flag) {
-    // Superclass constructor did locking
-    assert(monitor != nullptr, "null monitor not allowed");
-  }
+    MutexLocker(monitor, flag), _flag(flag) {}
```

Comment:
The assert will now be: `"null mutex is not allowed"`
instead of `"null monitor not allowed"`. Not really a
problem, just trying to make it clear.